### ISSUE TITLE
implemented interface is declared but its value is never read.

### ISSUE THREAD

**Issue by swaheed2** [palantir/tslint#3492]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.5.0
- __Running TSLint via__:  VSCode 

Ioinic :  3.9.0 

#### TypeScript code being linted

```ts
import { Component, Input, OnInit, OnDestroy } from '@angular/core';
export class PageBottom implements OnInit, OnDestroy {

	constructor(private appService: AppService) {
	}

	ngOnInit() {
		...
	}
       ngOnDestroy() {
		...
	}

}
```

with `tslint.json` configuration:

```
{
  "enable": true,
  "rules": {
    "no-duplicate-variable": true,
    "no-unused-variable": [
      true
    ]
  },
  "rulesDirectory": [
    "node_modules/tslint-eslint-rules/dist/rules"
  ],
  "autoFixOnSave": true
}
```

#### Actual behavior

'OnInit' is declared but its value is never read.

#### Expected behavior

'OnDestroy' is declared but its value is never read.

**Comment by ajafff** [comment]
Looks like a duplicate of #3463

Can you try upgrading to typescript@2.6? That fixed a similar issue.

**Comment by swaheed2** [comment]
Updated to 2.6.1 and still the same warnings

**Comment by ajafff** [comment]
Do you get the same errors when you enable `compilerOptions.noUnusedLocals` in your tsconfig.json?

**Comment by swaheed2** [comment]
yes, it looks like it's not recognizing that I have implemented OnInit.

**Comment by ajafff** [comment]
That might be a bug in TypeScript. `no-unused-variable` uses TypeScript's implementation of `--noUnusedLocals` and `--noUnusedParameters` behind the scenes.
If you can narrow this down to a self-contained reproduction, I suggest reporting it in the TypeScript repository.

**Comment by arush** [comment]
This is a bug for me too

TSLint version: 5.8.0
TypeScript version: 2.6.2
Running TSLint via: Node.js API

**Comment by mentatxx** [comment]
Same for me 

TSLint version: 5.9.1
TypeScript version: 2.6.2
Running TSLint via: Node.js API
